src/common/types/uint128_t.cpp

```diff
@@ -0,0 +1,740 @@
+#include "common/types/uint128_t.h"
+
+#include <cmath>
+
+#include "common/exception/overflow.h"
+#include "common/exception/runtime.h"
+#include "common/type_utils.h"
+#include "common/types/int128_t.h"
+#include "function/cast/functions/numeric_limits.h"
+#include "function/hash/hash_functions.h"
+// #include <concurrencysal.h>
+
+namespace kuzu::common {
+
+// NOLINTNEXTLINE(cert-err58-cpp): This initialization won't actually throw.
+const uint128_t Uint128_t::powerOf10[]{
+    uint128_t((uint64_t)1),
+    uint128_t((uint64_t)10),
+    uint128_t((uint32_t)100),
+    uint128_t((uint64_t)1000),
+    uint128_t((uint64_t)10000),
+    uint128_t((uint64_t)100000),
+    uint128_t((uint64_t)1000000),
+    uint128_t((uint64_t)10000000),
+    uint128_t((uint64_t)100000000),
+    uint128_t((uint64_t)1000000000),
+    uint128_t((uint64_t)10000000000),
+    uint128_t((uint64_t)100000000000),
+    uint128_t((uint64_t)1000000000000),
+    uint128_t((uint64_t)10000000000000),
+    uint128_t((uint64_t)100000000000000),
+    uint128_t((uint64_t)1000000000000000),
+    uint128_t((uint64_t)10000000000000000),
+    uint128_t((uint64_t)100000000000000000),
+    uint128_t((uint64_t)1000000000000000000),
+    uint128_t((uint64_t)1000000000000000000) * uint128_t((uint64_t)10),
+    uint128_t((int64_t)1000000000000000000) * uint128_t((int64_t)100),
```

Comment:
Repeat for lines below too
```suggestion
    uint128_t((uint64_t)1000000000000000000) * uint128_t((uint64_t)100),
```